internal/testing/e2e/consul.go

```diff
@@ -253,6 +256,24 @@ func consulCASecret(namespace string, caCert *testing.CertificateInfo) client.Ob
 	}
 }
 
+func consulGRPCVarName() string {
+	tagTokens := strings.Split(consulImage, ":")
+	tag := tagTokens[len(tagTokens)-1]
+	imageVersion, err := semver.NewVersion(tag)
+	if err != nil {
+		return "grpc"
+	}
+	breakingVersion, err := semver.NewVersion(grpcConsulIncompatibleNameVersion)
+	if err != nil {
+		return "grpc"
+	}
+	// we check major/minor directly since the semver check fails with the trailing -dev tag
+	if breakingVersion.Major() > imageVersion.Major() || (breakingVersion.Major() == imageVersion.Major() && breakingVersion.Minor() > imageVersion.Minor()) {
+		return "grpc"
+	}
+	return "grpc_tls"
+}
```

Comment:
This hack is due to the breaking change introduced in https://github.com/hashicorp/consul/commit/b9a9e1219c67778ff6662a494ed5482c088fe798#diff-d0522c3ebbb7255ee8b82416d7f46bbb05093ff8e25cde2000a46bd77189f7d7R6. Specifically the parameter name for gRPC configuration that's TLS-based changed from just `grpc` to `grpc_tls` and the ports for plaintext and TLS-based connections were split up.

Since we leverage Consul configured to use TLS in our e2e tests, without specifying the parameter in our testing consul config as `grpc_tls` on the latest consul images, then our test gateways wind up trying to use TLS on a plaintext connection and all xDS breaks. We'll need to ensure that this is handled properly on the `consul-k8s` [side of things](https://github.com/hashicorp/consul-k8s/blob/34c5774c2c53f7fca7f0819c7b31bd341dcc72ba/charts/consul/templates/api-gateway-gatewayclassconfig.yaml#L28-L29) since pinning the gateways xDS ports to 8502 when TLS is enabled in the cluster means we'll be breaking things, talking to a plaintext connection with TLS.